signalling/src/handlers/mod.rs

```diff
@@ -0,0 +1,1081 @@
+use webrtcsink_protocol as p;
+use anyhow::{anyhow, Error};
+use futures::prelude::*;
+use futures::ready;
+use pin_project_lite::pin_project;
+use std::collections::{HashMap, HashSet, VecDeque};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tracing::{info, instrument};
+
+pin_project! {
+    #[must_use = "streams do nothing unless polled"]
+    pub struct Handler {
+        #[pin]
+        stream: Pin<Box<dyn Stream<Item=(String, Option<p::IncomingMessage>)> + Send>>,
+        items: VecDeque<(String, p::OutgoingMessage)>,
+        producers: HashMap<String, HashSet<String>>,
+        consumers: HashMap<String, Option<String>>,
+        listeners: HashSet<String>,
```

Comment:
I think you at least want to add a `pub type PeerId = String` so all these funny strings everywhere have some more meaning.

Reply:
those fields are not public, but the type alias is nicer yes, done (without pub)